pkg/server/types.go

```diff
@@ -46,6 +50,33 @@ type HarvesterServer struct {
 	Handler http.Handler
 }
 
+const (
+	RancherKubeConfigSecretName = "rancher-kubeconfig"
+	RancherKubeConfigSecretKey  = "kubernetes.kubeconfig"
+)
+
+func RancherRESTConfig(ctx context.Context, restConfig *restclient.Config, options config.Options) (*restclient.Config, error) {
+	clientSet, err := kubernetes.NewForConfig(restConfig)
+	if err != nil {
+		return nil, err
+	}
+
+	secret, err := clientSet.CoreV1().Secrets(options.Namespace).Get(ctx, RancherKubeConfigSecretName, metav1.GetOptions{})
+	if err != nil {
+		if apierrors.IsNotFound(err) {
+			return restConfig, nil
+		}
```

Comment:
why not return the error instead?

Reply:
if user doesn't create secret rancher-kubeconfig, will use in-cluster kubeconfig instead.